### PR TITLE
Domain Upsell: Update the My home design

### PR DESCRIPTION
**Pull request by lupus2k**
Related to https://github.com/Automattic/dotcom-forge/issues/2207

## Proposed Changes

Update the Domain upsell Design on `/home` to match the Figma.
We also removed the `strike` on the current domain name on `/me` as commented here (pebzTe-RL-p2#comment-1694)
As now `/home` and `/me` have different designs, we create a new component for the profile page.

Figma: LruC8ztBCLJMFNkQN1JNm3-fi-2133_20773

As the PR got big because of the suggestions, I detailed the changes:
- [x] Move `/me` Domain Upsell to its own component
- [x] Remove all `/me` related data from the `/home` Domain Upsell component
- [x] Update styles on the `/home` Domain Upsell
- [x] Update e2e tests with suggestions from [here](https://github.com/Automattic/wp-calypso/pull/76326#discussion_r1178875347)

## Screenshots
| Desktop |  Mobile |
| --- | --- |
| image | image | 
| image | image |

#### `/me` design update
image

#### E2e
* First of all run `yarn build` on `/packages/calypso-e2e`
* Run `yarn workspace wp-e2e-tests test -- domain-upsell/domain-upsell__my-home`

#### Component
* For my home, run `yarn run test-client client/my-sites/customer-home/cards/features/domain-upsell/test/index.jsx`
* For profile, run `yarn run test-client client/me/domain-upsell/test/index.jsx`

## Testing Instructions

* Go to /home/:siteSlug
* With a site matching the conditions, you should see the Domain Upsell (email-verified + site without a custom domain + site setup not completed)
* Check the new design (desktop & mobile).
* Check `/me` to check the new design (desktop & mobile)

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~~
- [ ] ~~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~~
- [ ] ~~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~~

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/my-home-page.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/my-home-page.ts]: **suggestion** - there's a way to do this process of 1) checking whether a suggestion card is present, then 2) extracting the suggested domain out of that card, without using discouraged APIs.

GitHub doesn't allow me to select the entire method to make a suggestion, but the following is one example:

```suggestion
		const locator = this.page.locator( '.domain-upsell__card');
		
		await locator.waitFor();
		
		return await locator.locator('text').textContent();
```

Note that the suggested snippet sidesteps the use of `ElementHandle` API, [which has been discouraged by Playwright maintainers for some time](https://playwright.dev/docs/api/class-elementhandle). 

I've further changed the method to return whatever the text content is contained in in the target element, opting to leave the validation up to the caller, like this:

```typescript
// In a test step
const suggestedDomain = await myHomePage.suggestedDomain();
expect( suggestedDomain ).toBeTruthy();
```

Anyway, both are merely suggestions, but I would highly recommend the first suggestion to remove the use of ElementHandler API.

**Review by worldomonation — APPROVED** [review]
Although I left two suggestions, I will approve the change in this PR to unblock (since I will be AFK on 2023/04/28).

File: test/e2e/specs/domain-upsell/domain-upsell__my-home.ts
Comment (lupus2k) [inline test/e2e/specs/domain-upsell/domain-upsell__my-home.ts]: I'm not 100% sure of this.
I did it because `validateCartItem` expect a string, but `textContent()` could be `null`.

What do you think about it, @worldomonation, @dpasque ?

Comment (candy02058912) [reply test/e2e/specs/domain-upsell/domain-upsell__my-home.ts]:
```suggestion
		selectedDomain = suggestedDomain ?? '';
```
Suggesting a shorter solution: When `suggestedDomain` is `null` it will be the empty string, otherwise the `suggestedDomain`.

File: client/me/domain-upsell/index.jsx
Comment (candy02058912) [inline client/me/domain-upsell/index.jsx]:
```suggestion
	const shouldNotShowUpsellDismissed = ! hasPreferences || isDismissed;
```
typo

File: client/me/domain-upsell/index.jsx
Comment (candy02058912) [inline client/me/domain-upsell/index.jsx]:
```suggestion
		shouldNotShowUpsellDismissed ||
```

File: client/me/domain-upsell/index.jsx
Comment (candy02058912) [inline client/me/domain-upsell/index.jsx:90]: Thought that we could also add this `domainSuggestionOptions` in `client/my-sites/customer-home/cards/tasks/domain-upsell/index.jsx` in another PR. So we can change
```
// client/my-sites/customer-home/cards/tasks/domain-upsell/index.jsx
// Get first non-free suggested domain.
	const domainSuggestion = allDomainSuggestions?.filter(
		( suggestion ) => ! suggestion.is_free
	)[ 0 ];
```
to
```
// client/my-sites/customer-home/cards/tasks/domain-upsell/index.jsx
const domainSuggestion = allDomainSuggestions?.[ 0 ];
```
as well.

**Review by candy02058912 — APPROVED** [review]
Great work on doing this! I left some comments, but they're mostly minor.

✅ e2e tests passing locally
✅ Component tests passing locally
✅ tested /me 
me
✅ tested /home/:siteSlug
home

File: test/e2e/specs/domain-upsell/domain-upsell__my-home.ts
Comment (dpasque) [reply test/e2e/specs/domain-upsell/domain-upsell__my-home.ts]: I think that looks just fine, and I really like Candy's shorthand version!

Although, you probably don't need the `expect` and the fallback. If `suggestedDomain` is `null`, the `expect` call with throw and exit early! 👍

Comment (lupus2k) [reply test/e2e/specs/domain-upsell/domain-upsell__my-home.ts]: Thanks, @candy02058912, for the suggestion.
@dpasque, I thought the same, but without it, it fails at TeamCity because `cartCheckoutPage.validateCartItem` expect a string.

File: client/me/domain-upsell/index.jsx
Comment (lupus2k) [reply client/me/domain-upsell/index.jsx:90]: Thanks @candy02058912!

File: test/e2e/specs/domain-upsell/domain-upsell__my-home.ts
Comment (dpasque) [reply test/e2e/specs/domain-upsell/domain-upsell__my-home.ts]: Huh... Weird... 🤔  I think what you have is fine then! 👍

File: client/my-sites/customer-home/cards/features/domain-upsell/index.jsx
Comment (paulopmt1) [inline client/my-sites/customer-home/cards/features/domain-upsell/index.jsx:116]: Thanks for making them as a complete string, easy to find!

**Review by paulopmt1 — APPROVED** [review]
Good job @lupus2k! Loved to see these amount of tests being created!

- Should we reduce this line-height for mobile?

image

- Other than that:

  - ⛔ e2e tests not passing locally, can you please double check if that's the case? (sent you and inbox). If not, we're good to go 👍 
  - ✅ Component tests passing locally
  - ✅ tested /me 
  - ✅ tested /home/:siteSlug

File: test/e2e/specs/domain-upsell/domain-upsell__my-home.ts
Comment (worldomonation) [inline test/e2e/specs/domain-upsell/domain-upsell__my-home.ts:49]: **praise** - nice!

**Review by worldomonation — APPROVED** [review]
Approved from KitKat with regards to the changes in `test/e2e` and `packages/calypso-e2e`.

**Pull request merged**
